src/backend/distributed/metadata/distobject.c

```diff
@@ -254,6 +255,41 @@ IsObjectDistributed(const ObjectAddress *address)
 }
 
 
+/*
+ * ClusterHasDistributedFunction returns true if there is at least one distributed
+ * function in the cluster.
+ */
+bool
+ClusterHasDistributedFunction(void)
+{
+	Relation pgDistObjectRel = NULL;
+	SysScanDesc pgDistObjectScan = NULL;
+	HeapTuple pgDistObjectTup = NULL;
+
+	pgDistObjectRel = heap_open(DistObjectRelationId(), AccessShareLock);
+	pgDistObjectScan = systable_beginscan(pgDistObjectRel, InvalidOid, false, NULL, 0,
```

Comment:
Not a lot, unless you have a ton of schemas.

We do have an index (primary key):
```
postgres=# \d citus.pg_dist_object
                       Table "citus.pg_dist_object"
┌─────────────────────────────┬─────────┬───────────┬──────────┬─────────┐
│           Column            │  Type   │ Collation │ Nullable │ Default │
├─────────────────────────────┼─────────┼───────────┼──────────┼─────────┤
│ classid                     │ oid     │           │ not null │         │
│ objid                       │ oid     │           │ not null │         │
│ objsubid                    │ integer │           │ not null │         │
│ type                        │ text    │           │          │         │
│ object_names                │ text[]  │           │          │         │
│ object_args                 │ text[]  │           │          │         │
│ distribution_argument_index │ integer │           │          │         │
│ colocationid                │ integer │           │          │         │
└─────────────────────────────┴─────────┴───────────┴──────────┴─────────┘
Indexes:
    "pg_dist_object_pkey" PRIMARY KEY, btree (classid, objid, objsubid)
```

Reply:
Hmm, that's true. It looks like `systable_beginscan ` expects all the index columns to scankeys:
```C
 * If we choose to index, we reset them to 1..n to reference the index
 * columns.  Note this means there must be one scankey qualification per
 * index column! 
```

But, reading the `systable_beginscan` code, we may only pass `classid `, since it is the first column in the index and it should(?) work fine. Though, we probably wouldn't want to rely on that?

Reply:
It can probably, but seems fine to leave it as is.